compiler/rustc_trait_selection/src/traits/object_safety.rs

```diff
@@ -838,9 +838,10 @@ fn contains_illegal_self_type_reference<'tcx, T: TypeFoldable<'tcx>>(
                         let leaf = leaf.subst(self.tcx, ct.substs);
                         self.visit_const(leaf)
                     }
-                    Node::Binop(..) | Node::UnaryOp(..) | Node::FunctionCall(_, _) => {
-                        ControlFlow::CONTINUE
-                    }
+                    Node::Binop(..)
+                    | Node::UnaryOp(..)
+                    | Node::FunctionCall(_, _)
+                    | Node::Cast(_, _, _) => ControlFlow::CONTINUE,
```

Comment:
same thing here about `visit_ty`

Reply:
yea, it's usually strictly safer to just visit and let monomorphization clean up if that's unnecessary